lib/api/model.js

```diff
@@ -15,6 +17,101 @@ function get (modelName, passByReference) {
     return clone(linz.get('models')[modelName]);
 }
 
+/**
+ * Get permission data for a model
+ * @param {Object}   req      Request data
+ * @param {String}   modelName Model name
+ * @param {Function} cb        Callback function
+ * @return {Object}
+ */
+function getPermissions(req, modelName, cb) {
+
+    if (!linz.get('disabled permission cache') && !req.linz.cache.permissions.invalidate && req.session.permissions && req.session.permissions[modelName]) {
+        return cb(null, req.session.permissions[modelName]);
+    }
+
+    var model = get(modelName);
+
+    var permissions = model.linz.formtools.grid.actions.concat(
+            model.linz.formtools.grid.recordActions,
+            model.linz.formtools.grid.groupActions,
+            model.linz.formtools.overview.actions,
+            model.linz.formtools.grid.export
+        )
+        .filter(function (action) {
+            return Object.keys(action).indexOf('permission') >= 0;
+        })
+        .map(function (action) {
+            return action.permission;
+        })
+        .concat(['index','create','overview','edit','delete','json']);
+
+    async.seq(
```

Comment:
As discussed, we agreed there's no need to have a 'default' detection since we have caching in place so this bit of code will run once for every new session.